lib/Dashboard/TalkWidget.php

```diff
@@ -200,8 +202,8 @@ protected function prepareRoom(Room $room, string $userId): WidgetItem {
 
 		if ($room->getCallFlag() !== Participant::FLAG_DISCONNECTED) {
 			$subtitle = $this->l10n->t('Call in progress');
-		} elseif ($participant->getAttendee()->getLastMentionMessage() > $participant->getAttendee()->getLastReadMessage()) {
-			$subtitle = $this->l10n->t('You were mentioned');
+		} elseif ($participant->getAttendee()->getLastMentionDirect() > $participant->getAttendee()->getLastReadMessage()) {
+			$subtitle = $this->chatManager->getComment($room, (string)$participant->getAttendee()->getLastMentionDirect())->getMessage();
```

Comment:
Added the checks for expired/deleted messages. And, also added the parse function to display the correct mention name instead of @mentionid. 
I think similar checks/changes will be required in RoomFormatter.php for lastUnreadDirectMention, I will look at it tomorrow.

Reply:
Similar checks and parsing have also been implemented for RoomFormatter to check deleted or expired messages. Also, the actual name of the users are displayed instead of @userId.